src/test/java/nextstep/subway/unit/entity/LineTest.java

```diff
@@ -172,4 +172,80 @@ public void sut_split_distance_if_middle_section() {
             assertThat(sut.getSections().get(1).getDistance()).isEqualTo(6L);
         }
     }
+
+    @DisplayName("deleteSection")
+    @Nested
+    class DeleteSection {
+        @Test
+        public void sut_throws_if_section_size_one() {
+            // given
+            Line sut = LineFixture.prepareLineOne(1L, 2L);
+
+            // when
+            SubwayDomainException actual = (SubwayDomainException) catchThrowable(() -> sut.deleteSection(1L));
+
+            // then
+            assertThat(actual.getExceptionType()).isEqualTo(SubwayDomainExceptionType.INVALID_SECTION_SIZE);
+        }
+
+        @Test
+        public void sut_throws_if_not_includes_station() {
+            // given
+            Line sut = LineFixture.prepareLineOne(1L, 4L);
+
+            // when
+            SubwayDomainException actual = (SubwayDomainException) catchThrowable(() -> sut.deleteSection(999L));
+
+            // then
+            assertThat(actual.getExceptionType()).isEqualTo(SubwayDomainExceptionType.INVALID_STATION);
+        }
+
+        @Test
+        public void sut_delete_section_if_first_section() {
+            // given
+            Line sut = LineFixture.prepareLineOne(1L, 4L);
+
+            // when
+            sut.deleteSection(1L);
+
+            // then
+            assertThat(sut.getSections().getAllStationIds()).isEqualTo(List.of(2L, 3L, 4L));
+        }
+
+        @Test
+        public void sut_delete_section_if_last_section() {
+            // given
+            Line sut = LineFixture.prepareLineOne(1L, 4L);
+
+            // when
+            sut.deleteSection(4L);
+
+            // then
+            assertThat(sut.getSections().getAllStationIds()).isEqualTo(List.of(1L, 2L, 3L));
+        }
+
+        @Test
+        public void sut_delete_section_if_middle_section() {
+            // given
+            Line sut = LineFixture.prepareLineOne(1L, 4L);
+
+            // when
+            sut.deleteSection(2L);
+
+            // then
+            assertThat(sut.getSections().getAllStationIds()).isEqualTo(List.of(1L, 3L, 4L));
```

Comment:
```suggestion
            Line sut = LineFixture.prepareLine(강남역, 선릉역, 삼성역, 잠실역);

            // when
            sut.deleteSection(선릉역);

            // then
            assertThat(sut.getSections().getAllStationIds()).isEqualTo(List.of(강남역, 삼성역, 잠실역));
```
기구현된 `prepareLineOne()`에서 for문을 통해 역을 생성하는데 가독성을 높이기 위해 
- 추가하는 Station을 알기 쉽게 가변인자를 사용
-  StationFixture를 매개변수로 사용

해보는 것도 괜찮을 것 같아요!

Reply:
제가 Station 을 직접참조하지 않고 stationId 를 간접 참조하도록 개발해서 Fixture 로 빼기에 애매한 부분이 있습니다.
그렇다고 인수테스트처럼 한글 = id (Long) 으로 놓기에는 오히려 더 헷갈리는거 같습니다.

혹은 내부에서는 간접참조하더라도 Line / LineSection 생성시 Station 을 직접 넘기도록 생성자를 구성할까 생각도 해보았는데... 
테스트를 위해서 바꾸는게 맞나? 생각이 들었습니다. 혹시 어떻게 하는게 좋아보일까요?

Reply:
음, 간접 참조일 경우는 말씀하신 내용처럼 Fixture 구성이 애매할 수 있겠네요.. 

> 그래서 내부에서는 간접참조하더라도 Line / LineSection 생성시 Station 을 직접 넘기도록 생성자를 구성할까 생각도 해보았는데... 테스트를 위해서 바꾸는게 맞나? 생각이 들었습니다.

저는 Line과 Section 은 같은 라이프사이클을 공유하고 있다고 생각해 직접 참조가 더 적합하다고 생각합니다. 
경석님은 간접 참조를 선택하신 이유가 어떤건가요?

Reply:
저도 Line 과 Section 은 같은 라이프 사이클이라고 생각했습니다!

하지만 Station 의 경우는 Line 과 Section 이 없더라도 독립적으로 존재할 수 있기 때문에 간접참조로 설정했습니다.

Reply:
그럼 소스 수정 없이 유지하는걸로 가시죠😀